clients/js/src/signed_calls.ts

```diff
@@ -21,8 +18,68 @@ const DEFAULT_GAS_LIMIT = 30_000_000;
 const DEFAULT_VALUE = 0;
 const DEFAULT_DATA = '0x';
 const zeroAddress = () => `0x${'0'.repeat(40)}`;
+const MAX_SAFE_BIG_NUM = 0x1fffffffffffff - 1;
 
-export type Signer = Pick<EthersSigner, 'getTransactionCount' | 'getChainId'> &
+class SignedCallCacheManager {
+  public enabled = true;
+
+  // for each signer, we cache the signature of the hash of each SignableCall
+  private cachedSignatures = new Map<string, Map<string, Uint8Array>>();
+  // for each ChainId, we cache the base block number to make the same leash
+  private cachedBlockNumbers = new Map<number, BlockTag>();
+
+  public enable() {
+    this.enabled = true;
+  }
+
+  public disable() {
+    this.enabled = false;
+    cachedSignatures.clear();
+    cachedBlockNumbers.clear();
+  }
+
+  public setSignatureCache(
+    address: string,
+    hash: string,
+    signature: Uint8Array,
+  ) {
```

Comment:
```suggestion
  public set(
    dataPack: SignedCallDataPack,
    signature: Uint8Array
  ) {
```
having the API like this frees you to use any method of digesting the data pack into a map key